ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos/Common.hs

```diff
@@ -54,36 +55,178 @@ data PraosChainSelectView c = PraosChainSelectView
   }
   deriving (Show, Eq, Generic, NoThunks)
 
+-- | When to compare the VRF tiebreakers.
+data VRFTiebreakerFlavor =
+    -- | Always compare the VRF tiebreakers. This is the behavior of all eras
+    -- before Conway. Once mainnet has transitioned to Conway, we can remove
+    -- this option.
+    UnrestrictedVRFTiebreaker
+  | -- | Only compare the VRF tiebreakers when the slot numbers differ by at
+    -- most the given number of slots.
+    --
+    -- The main motivation is as follows:
+    --
+    -- When two blocks A and B with the same block number differ in their slot
+    -- number by more than Δ (the maximum message delay from Praos), say
+    -- @slot(A) + Δ < slot(B)@, the issuer of B should have been able to mint a
+    -- block with a block number higher than A (eg by minting on top of A).
+    -- Therefore, we do not want to allow B to win against A by having a better
+    -- VRF tiebreaker, such that properly configured pools (like the issuer of
+    -- A) do not lose blocks because of poorly configured pools (like the issuer
+    -- of B).
+    RestrictedVRFTiebreaker SlotNo
+  deriving stock (Show, Eq, Generic)
+  deriving anyclass (NoThunks)
+
+-- Used to implement the 'Ord' and 'ChainOrder' instances for Praos.
+comparePraos ::
+  Crypto c =>
+  VRFTiebreakerFlavor ->
+  PraosChainSelectView c ->
+  PraosChainSelectView c ->
+  Ordering
+comparePraos tiebreakerFlavor =
```

Comment:
How about this instead?

```
  mconcat
    [ compare `on` csvChainLength,
      when' ((==) `on` csvIssuer) (compare `on` csvIssueNo),
      when' vrfArmed  (compare `on` Down . csvTieBreakVRF)
    ]  where
    -- When the given condition is satisfied, use the given comparison
    -- function, otherwise, no preference.
    when' b comp v1 v2 = if b v1 v2 then comp v1 v2 else EQ

    vrfArmed v1 v2 = case tiebreakerFlavor of
        UnrestrictedVRFTiebreaker       -> True
        RestrictedVRFTiebreaker maxDist ->
            slotDist (csvSlotNo v1) (csvSlotNo v2) <= maxDist

    slotDist :: SlotNo -> SlotNo -> SlotNo
    slotDist s t
      -- slot numbers are unsigned, so have to take care with subtraction
      | s >= t    = s - t
      | otherwise = t - s
```

Reply:
And, more of a personal preference, since we're in the neighborhood :)

```
  (compare `on` csvChainLength)
    <> when' ((==) `on` csvIssuer) (compare `on` csvIssueNo)
    <> when' vrfArmed  (compare `on` Down . csvTieBreakVRF)
```

Reply:
Yeah, moving this function around seems like a good opportunity to condense the logic :+1: